pool/app/query/entity.ml

```diff
@@ -0,0 +1,156 @@
+module Common = Pool_common
+module Message = Common.Message
+module Dynparam = Utils.Database.Dynparam
+
+module Pagination = struct
+  module Limit = struct
+    include Common.Model.Integer
+
+    let default = 20
+    let field = Message.Field.Limit
+    let create m = if m >= 0 then Ok m else Error (Message.Invalid field)
+    let schema = schema field create
+  end
+
+  module Page = struct
+    include Common.Model.Integer
+
+    let default = 1
+    let field = Message.Field.Page
+    let create m = if m > 0 then Ok m else Error (Message.Invalid field)
+    let schema = schema field create
+  end
+
+  module PageCount = struct
+    include Common.Model.Integer
+
+    let field = Message.Field.PageCount
+    let create m = if m >= 0 then Ok m else Error (Message.Invalid field)
+    let schema = schema field create
+  end
+
+  type t =
+    { limit : Limit.t
+    ; page : Page.t
+    ; page_count : PageCount.t
+    }
+  [@@deriving eq, show]
+
+  let create ?limit ?page ?(page_count = 1) () =
+    let open CCOption in
+    let open CCFun in
+    let get_value = value in
+    let build input create default =
+      input |> map_or ~default (create %> of_result %> get_value ~default)
```

Comment:
Use `value` directly (if it's something with the name, you can use the `get_or` function as well)

```suggestion
      input |> map_or ~default (create %> of_result %> value ~default)
```